drivers/gpio/sabertooth_analog_driver.go

```diff
@@ -0,0 +1,294 @@
+    SetMaxPower(ratio float64) error
+}
+
+// NewSabertoothAnalogDriver creates a new SabertoothAnalogDriver given a PwmWriter,
+// the DIP switch settings and the pins connected to the S1, S2, A1 and A2 terminals
+// of the driver.
+//
+// Adds the following API commands:
+//  throttle(value float64) - see SabertoothAnalogDriver.Throttle
+//  differential(value float64) - see SabertoothAnalogDriver.Differential
+func NewSabertoothAnalogDriver(w PwmWriter, d SabertoothDIPSwitches, s1, s2, a1, a2 string) SabertoothAnalogDriver {
+    s := &sabertoothAnalogDriver{
+        name:       gobot.DefaultName("SabertoothAnalogDriver"),
+        switches:   d,
+        s1:         s1,
+        s2:         s2,
+        a1:         a1,
+        a2:         a2,
+        m1Ratio:    1,
+        m2Ratio:    1,
+        connection: w,
+        Commander:  gobot.NewCommander(),
+    }
+    
+    s.AddCommand("throttle", func(params map[string]interface{}) interface{} {
+        return s.Throttle(params["value"].(float64))
+    })
+    s.AddCommand("differential", func(params map[string]interface{}) interface{} {
+        return s.Differential(params["value"].(float64))
+    })
+    
+    return s
+}
+
+func (s *sabertoothAnalogDriver) Start() error { return nil }
+
+func (s *sabertoothAnalogDriver) Halt() error { return nil }
+
+func (s *sabertoothAnalogDriver) Name() string { return s.name }
+
+func (s *sabertoothAnalogDriver) SetName(n string) { s.name = n }
+
+func (s *sabertoothAnalogDriver) Connection() gobot.Connection {
+    return s.connection.(gobot.Connection)
+}
+
+func (s *sabertoothAnalogDriver) PinS1() string { return s.s1 }
+
+func (s *sabertoothAnalogDriver) PinS2() string { return s.s2 }
+
+func (s *sabertoothAnalogDriver) PinA1() string { return s.a1 }
+
+func (s *sabertoothAnalogDriver) PinA2() string { return s.a2 }
+
+func (s *sabertoothAnalogDriver) GetThrottle() float64 { return s.throttle }
+
+func (s *sabertoothAnalogDriver) GetDifferential() float64 { return s.differential }
+
+func (s *sabertoothAnalogDriver) write(pin string, value float64) error {
+    if pin != "" {
+        value = math.Min(math.Max(value, 0), 1)
+        value = gobot.Rescale(value, 0, 1, 0, math.MaxUint8)
+        return s.connection.PwmWrite(pin, byte(math.Round(value)))
+    }
+    return nil
+} // END write
+
+func (s *sabertoothAnalogDriver) SetRampRate(rate float64) error {
+    if !s.switches.Bidirectional() { return ErrBidirectionalOperation }
+    if s.a1 == "" { return ErrA1NotSet }
+    rate = math.Min(math.Max(rate, 0), 1)
+    return s.write(s.a1, rate)
+} // END SetRampRate
+
+func (s *sabertoothAnalogDriver) SetMaxPower(ratio float64) error {
+    if !s.switches.Bidirectional() { return ErrBidirectionalOperation }
+    if s.a2 == "" { return ErrA2NotSet }
+    ratio = math.Min(math.Max(ratio, 0), 1)
+    return s.write(s.a2, ratio)
+} // END SetMaxPower
+
+func (s *sabertoothAnalogDriver) Throttle(throttle float64) error {
```

Comment:
This function looks quite complicate, so my suggestion is to add some unit test.